pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -1010,6 +1018,27 @@ func getJmsConfig(environment api.EnvironmentType, jms *api.KieAppJmsObject) (*a
 	return jms, nil
 }
 
+func getKafkaConfig(kafka *api.KafkaExtObject) (*api.KafkaExtObject, error) {
+	if kafka == nil || !kafka.ExtEnabled {
+		return nil, nil
+	}
+
+	defaultKafka := api.KafkaExtObject{
+		MaxBlockMs:       200,
+		AutocreateTopics: Pbool(true),
+		BootstrapServers: "localhost:9092",
+		GroupID:          "jbpm-consumer",
+		Acks:             2,
+	}
```

Comment:
are we not allowing the user to set everything via the cr spec? it appears you're setting defaults and its not configurable?

Reply:
Removed

Reply:
i didn't mean to remove it... instead there should be logic that checks if the field is blank before setting a default. if its not blank, then the current setting should be used. right?

Reply:
Added, we set missing values and override if something has changed